qcodes/instrument/parameter.py

```diff
@@ -1349,35 +1381,89 @@ def _update_with(self, *,
         def __call__(self) -> ParamDataType:
             return self.get(get_if_invalid=True)
 
-    def __init__(self, name: str, source: Parameter, *args: Any,
+    def __init__(self, name: str, source: Optional[Parameter], *args: Any,
                  **kwargs: Any):
-        self.source = source
 
-        for ka, param in zip(('unit', 'label', 'snapshot_value'),
-                             ('unit', 'label', '_snapshot_value')):
-            kwargs[ka] = kwargs.get(ka, getattr(self.source, param))
+        self._attr_inherit = {"label": {"fixed": False,
+                                        "default_none_value": name},
+                              "unit": {"fixed": False,
+                                       "default_none_value": ""}}
+
+        for attr, attr_props in self._attr_inherit.items():
+            if attr in kwargs:
+                attr_props["fixed"] = True
+            else:
+                attr_props["fixed"] = True
```

Comment:
typo-bug?
```suggestion
                attr_props["fixed"] = False
```

Reply:
Hmn yes. Seems like the tests are not strict enough then

Reply:
Fixed and added tests